cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -159,10 +159,19 @@ bool SubstraitToVeloxPlanValidator::validateRegexExpr(
     LOG_VALIDATION_MSG("Pattern is not string literal for " + name);
     return false;
   }
-
   const auto& pattern = patternArg.literal().string();
+
+  std::string rewrite;
+  if (name == "regexp_replace " && scalarFunction.arguments().size() > 2) {
+    const auto& rewriteArg = scalarFunction.arguments()[2].value();
+    if (!rewriteArg.has_literal() || !rewriteArg.literal().has_string()) {
+      LOG_VALIDATION_MSG("Rewrite is not string literal for " + name);
+      return false;
```

Comment:
This check will make this function always fall back if non-constant replacement string is used.

Please check whether this comment makes sense: https://github.com/apache/incubator-gluten/issues/6224#issuecomment-2207890612.

Reply:
@kecookier, if the proposal is feasible, you can file a drafted pr in velox to see whether the community accepts that. Thanks!